netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -89,7 +89,7 @@ public static ProtocolNegotiator serverPlaintext() {
     return new ProtocolNegotiator() {
       @Override
       public Handler newHandler(final Http2ConnectionHandler handler) {
-        return new Handler() {
+        class PlaintextHandler extends ChannelDuplexHandler implements Handler {
```

Comment:
Handler is an interface, which most instances are of ChannelDuplexHandler.  Handler also has a scheme() method, so it can't return an anonymous class (can't both extend and implement).  I tried changing halnder over to ChannelInboundHandler, but then that messes up our instances.